test/integration/install_test.go

```diff
@@ -1248,6 +1248,20 @@ func TestRestarts(t *testing.T) {
 	}
 }
 
+// TestCleanUp deletes the resources used in the above tests
+func TestCleanUp(t *testing.T) {
```

Comment:
Instead of adding this cleanup test, could we make a similar change as we are in `opaque_ports_test.go` and create the namespaces with `TestHelper.WithDataPlaneNamespace` [here](https://github.com/linkerd/linkerd2/blob/e219ee0148532bbf0dd82d459a9af528aec76532/test/integration/install_test.go#L1132)?

Reply:
@kleimkuhler I don't think we can do that here because there[ are other Test functions in this suite ](https://github.com/linkerd/linkerd2/blob/e219ee0148532bbf0dd82d459a9af528aec76532/test/integration/install_test.go#L1204)that also use these namespaces. Hence it had to be separated to be at the end. WDYT?

Reply:
Okay. I do think its a good idea to keep the tests more independent, but since this is how it already works and we're just adding the cleanup, then I'm good with this change.

Maybe its worth looking into separating these out into their own `TestHelper.WithDataPlaneNamespace` tests, but not as part of this change.